app/packages/core/src/components/Actions/DynamicGroupAction/useDynamicGroupChoices.ts

```diff
@@ -0,0 +1,35 @@
+import * as fos from "@fiftyone/state";
+import { getFetchFunction } from "@fiftyone/utilities";
+import { useEffect, useState } from "react";
+import { useRecoilValue } from "recoil";
+
+export function useDynamicGroupChoices() {
+  const datasetName = useRecoilValue(fos.datasetName);
+  const [isLoading, setIsLoading] = useState(false);
+  const [availableFields, setAvailableFields] = useState<string[] | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    (async () => {
+      setIsLoading(true);
+      try {
+        const res = await getFetchFunction()<
```

Comment:
Prefer GraphQL, but ok. But also, this data can be resolved client side against the schema, save for the field counts which can use the existing GraphQL API / aggregation selector